File: test/jiebatest.py
import jieba

if __name__ == '__main__':
    print(jieba.__version__)
    result = []
    jieba.add_word('__解释性__')
    with open("./example.txt",encoding='utf-8') as f:
        for line in f:
            result.append(list(jieba.cut(line.strip(),HMM=False)))

    with open("./py_ans.txt","w", encoding="utf-8") as f:
        for line in result:
            f.write("|".join(line)+"\n")

File: test/jieba_test.go
package test

import (
	"fmt"
	"time"
	"github.com/yat011/jiebagosame"
	"testing"
	"os"
	"bufio"
	"strings"
)

func TestJieba(t * testing.T) {

	tokenizer, err := jiebagosame.NewTokeniezer("../dict.txt")
	if err != nil{
		panic(err)
	}
	file, _ :=os.Open("./example.txt")
	defer file.Close()
	scanner := bufio.NewScanner(file)
	content := make([]string,0)
	for scanner.Scan(){
		line := scanner.Text()
		content = append(content, line)
	}


	result := make([][]string, 0)
	start := time.Now()
	for _ ,line := range content{
		result = append(result,tokenizer.Cut(line, false, false))
	}
	elapsed := time.Since(start)
	fmt.Printf("took %s\n", elapsed)
	//fmt.Println("hihi")

	ansFile, aErr := os.Create("./go_ans.txt")
	defer ansFile.Close()
	if aErr != nil{
		panic(aErr)
	}
	for _, ans := range result{
		fmt.Fprintf(ansFile,"%s\r\n", strings.Join(ans, "|"))
	}
}